content/knowledge-base/guides/write-a-composition-function-in-go.md

```diff
@@ -0,0 +1,826 @@
+flag tells the function to run without encryption or authentication. You should
+only use it during testing and development. The
+{{<hover label="run" line="1">}}--debug{{</hover>}} flag tells the function to
+print debug log statements.
+
+```shell {label="run"}
+go run . --insecure --debug
+```
+
+In a separate terminal, run `crossplane beta render`. 
+
+```shell
+crossplane beta render xr.yaml composition.yaml functions.yaml
+```
+
+This command calls your function. In the terminal where your function is running
+you should now see log output:
+
+```shell
+go run . --insecure --debug
+2023-10-31T16:17:32.158-0700    INFO    function-xbuckets/fn.go:29      Running Function        {"tag": ""}
+2023-10-31T16:17:32.159-0700    INFO    function-xbuckets/fn.go:125     Added desired buckets   {"xr-version": "example.crossplane.io/v1", "xr-kind": "XBuckets", "xr-name": "example-buckets", "region": "us-east-2", "count": 3}
+```
+
+The `crossplane beta render` command prints the desired resources the function
+returns.
+
+```yaml
+---
+apiVersion: example.crossplane.io/v1
+kind: XBuckets
+metadata:
+  name: example-buckets
+---
+apiVersion: s3.aws.upbound.io/v1beta1
+kind: Bucket
+metadata:
+  annotations:
+    crossplane.io/composition-resource-name: xbuckets-crossplane-functions-example-b
+    crossplane.io/external-name: crossplane-functions-example-b
+  generateName: example-buckets-
+  labels:
+    crossplane.io/composite: example-buckets
+  ownerReferences:
+    # Omitted for brevity
+spec:
+  forProvider:
+    region: us-east-2
+---
+apiVersion: s3.aws.upbound.io/v1beta1
+kind: Bucket
+metadata:
+  annotations:
+    crossplane.io/composition-resource-name: xbuckets-crossplane-functions-example-c
+    crossplane.io/external-name: crossplane-functions-example-c
+  generateName: example-buckets-
+  labels:
+    crossplane.io/composite: example-buckets
+  ownerReferences:
+    # Omitted for brevity
+spec:
+  forProvider:
+    region: us-east-2
+---
+apiVersion: s3.aws.upbound.io/v1beta1
+kind: Bucket
+metadata:
+  annotations:
+    crossplane.io/composition-resource-name: xbuckets-crossplane-functions-example-a
+    crossplane.io/external-name: crossplane-functions-example-a
+  generateName: example-buckets-
+  labels:
+    crossplane.io/composite: example-buckets
+  ownerReferences:
+    # Omitted for brevity
+spec:
+  forProvider:
+    region: us-east-2
+```
+
+{{<hint "tip">}}
+Read the composition functions documentation to learn more about
+[testing composition functions]({{<ref "../../master/concepts/composition-functions#test-a-composition-that-uses-functions">}}).
+{{</hint>}}
+
+## Build and push the function to a package registry
+
+You build a function in two stages. First you build the function's runtime. This
+is the Open Container Initiative (OCI) image Crossplane uses to run your
+function. You then embed that runtime in a package, and push it to a package
+registry. The Crossplane CLI uses `xpkg.upbound.io` as its default package
```

Comment:
@plumbis Vale is picking up xpkg.upbound.io as a spell check error here too. Is there a way to avoid that? Should I just add it to the dictionary?